go/mysql/auth_server_static.go

```diff
@@ -242,6 +231,8 @@ func (a *AuthServerStatic) ValidateHash(salt []byte, user string, authResponse [
 // We only recognize MysqlClearPassword and MysqlDialog here.
 func (a *AuthServerStatic) Negotiate(c *Conn, user string, remoteAddr net.Addr) (Getter, error) {
 	// Finish the negotiation.
+
+	// a.Entries is accessed within this function, so we lock here
```

Comment:
Yipes! I just noticed this now, but we are holding the lock while waiting for the negotiation packet from the client which is a recipe for potential disaster.

We really need to move _this_ lock after the negotiate but before the access to the Entries array.